msticpy/data/drivers/mordor_driver.py

```diff
@@ -0,0 +1,809 @@
+    """
+    md_idx_techniques = defaultdict(set)
+    md_idx_tactics = defaultdict(set)
+
+    for md_id, md_file in mdr_metadata.items():
+        for attack in md_file.get_attacks():
+            md_idx_techniques[attack.technique].add(md_id)
+            if not attack.tactics:
+                continue
+            for tactic in attack.tactics:
+                md_idx_tactics[tactic].add(md_id)
+
+    return md_idx_techniques, md_idx_tactics  # type: ignore
+
+
+def download_mdr_file(
+    file_uri: str, use_cached: bool = True, save_folder: str = ".", silent: bool = False
+) -> pd.DataFrame:
+    """
+    Download data file from Mordor.
+
+    Parameters
+    ----------
+    file_uri : str
+        The URI of the file to download.
+    use_cached : bool, optional
+        Try to use locally saved file first, by default True
+    save_folder : str, optional
+        Path to output folder, by default "."
+    silent : bool
+        If True, suppress feedback. By default, False.
+
+    Returns
+    -------
+    pd.DataFrame
+        DataFrame of Dataset
+
+    """
+    if not silent:
+        print(file_uri)
+    if not file_uri.lower().endswith("zip"):
+        raise TypeError(f"File type not supported {file_uri}")
+    save_path = "-".join(file_uri.split("/")[-2:-1])
+    save_file = Path(save_folder).joinpath(save_path)
+    if not use_cached or not save_file.is_file():
+        # streamed download
+        resp = requests.get(file_uri, stream=True)
+        with open(str(save_file), "wb") as fdesc:
+            for chunk in resp.iter_content(chunk_size=1024):
+                fdesc.write(chunk)
+
+    zip_file = zipfile.ZipFile(str(save_file))
+    file_names = zip_file.namelist()
+    d_frames = {}
+    for file_name in file_names:
+        d_frames[file_name] = _extract_zip_file_to_df(
+            zip_file, file_name, use_cached, save_folder, silent
+        )
+
+    return pd.concat(d_frames.values())
+
+
+def _extract_zip_file_to_df(  # noqa: MC0001
+    zip_file: ZipFile,
+    file_name: str,
+    use_cached: bool = True,
+    save_folder: str = ".",
+    silent: bool = False,
+) -> pd.DataFrame:
+    """
+    Extract from zip and parse json file to DataFrame.
+
+    Parameters
+    ----------
+    zip_file : ZipFile
+        ZipFile object containing the file
+    file_name : str
+        File name to extract
+    use_cached : bool, optional
+        Try to use locally saved file first, by default True
+    save_folder : str, optional
+        Path to output folder, by default "."
+    silent : bool
+        If False, suppress feedback. By default, True.
+
+    Returns
+    -------
+    pd.DataFrame
+        Extracted DataFrame
+
+    """
+    if not silent:
+        print("Extracting", file_name)
+
+    file_path = Path(save_folder).joinpath(file_name)
+    if not use_cached or not file_path.is_file():
+        zip_file.extract(file_name, path=save_folder)
+
+    out_df = pd.DataFrame()
+    if file_path.suffix.lower() == ".json":
```

Comment:
would it not be more efficent just to use pandas read_json() here?
https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.read_json.html?highlight=read_json

I tested this with a few of the morder datasets and pd.read_csv('file', line=True) seemed to work fine.

Reply:
Yep, you're right. I hadn't seen the lines parameter and couldn't get it to work.